### DISCUSSION TITLE
View the home timeline from multiple accounts

### DISCUSSION THREAD

**Discussion by nikclayton** [pachli/pachli-android#9] [Ideas]
https://qoto.org/@falken/111030751049060686 writes: "does it merge all accounts into a single timeline? Would be killer feature, all I miss in @Tusky"

The user would be able to create a new tab that contained posts from multiple "Home" timelines for their other accounts.

This option should only appear if they have multiple accounts configured.

When viewing this timeline they would see all the posts from the home timelines, and could interact with them as normal.

This poses a number of tricky UX questions that I don't have good answers for at the moment.

- Should each post in the timeline include a clear indicator for the account's timeline it was originally in?
   - If so, what does that indicator look like?
   - What if the post is in the home timeline for multiple accounts, how is that shown?
- If the user taps on a post to view it's thread, which of the user's accounts is used to fetch the thread?
   - The user's servers might have different views of the thread contents, because of how federation works
- If the user interacts with the post, how do they choose which of their logged in accounts they're going to use to interact with it?
   - Interaction being boost, favourite, bookmark, reply, share, pin, mute, block, ...
   - The user might have accounts on two instances that do not federate with each other
- If the user has alt accounts they want to keep private, how do we make sure this feature doesn't inadvertently cause the user to disclose the existence of the alt (e.g., by replying to a thread from account B when they meant to reply from account A)?

There's probably more, that's five minutes of thought.

Instead of merging all accounts in to one one timeline, perhaps allowing the user to add multiple tabs, each one containing the home timeline for a different account would be an acceptable compromise?

Thoughts and suggestions welcomed.

**Comment by complex-hub** [comment]
Just prefacing this by saying I don't have much experience with mobile dev, so I don't know the feasibility of implementing the solutions but here are my comments about these points you raised:

> - Should each post in the timeline include a clear indicator for the account's timeline it was originally in?

One way to do this could be by inserting the origin account profile picture below the poster's own profile picture. For posts that appear on more than one account, they could be de-duplicated and another indicator could represent this. Kinda like this:

  1. Post is only on one account: 
![single_user_tl](https://github.com/pachli/pachli-android/assets/53913349/841d2caa-b0cb-421d-9de7-1084b5578c90)

  2. Post is on multiple accounts (in this case 2):
![multi_user_tl](https://github.com/pachli/pachli-android/assets/53913349/45ca43d6-6dd8-4376-8eca-6a6adf34a7fb)

> - If the user taps on a post to view it's thread, which of the user's accounts is used to fetch the thread?

Similar to the main timeline, threads could be fetched using all the accounts, with the duplicate posts being removed/consolidated into a single one.

> - If the user interacts with the post, how do they choose which of their logged in accounts they're going to use to interact with it?

The setting "Show confirmation before boosting/favoriting" could be used for this. When more than one account is configured, liking and boosting a post would launch a pop-up with the accounts that can perform the action.

For a more seamless experience, an "account preference" setting could be implemented, similar to the current tabs settings, in which the user chooses the order that the app should attempt to perform the action:

![accounts_tab](https://github.com/pachli/pachli-android/assets/53913349/17397c94-09f6-4048-a58f-4eb55701fc8f)

> - If the user has alt accounts they want to keep private, how do we make sure this feature doesn't inadvertently cause the user to disclose the existence of the alt (e.g., by replying to a thread from account B when they meant to reply from account A)?

The current reply UI already addresses this point, no? Since the user can see on the top right which account is being used to send the post. Anyway, a confirmation could be added in cases when more than one account can reply to a message, just to add another step to prevent accidental replies.
